// @aws-sdk/client-ecr/dist/es/commands/DeleteLifecyclePolicyCommand.js
import { __extends } from "tslib";
import { DeleteLifecyclePolicyRequest, DeleteLifecyclePolicyResponse } from "../models/models_0";
import { deserializeAws_json1_1DeleteLifecyclePolicyCommand, serializeAws_json1_1DeleteLifecyclePolicyCommand, } from "../protocols/Aws_json1_1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Deletes the lifecycle policy associated with the specified repository.</p>
 */
var DeleteLifecyclePolicyCommand = /** @class */ (function (_super) {
    __extends(DeleteLifecyclePolicyCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DeleteLifecyclePolicyCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DeleteLifecyclePolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ECRClient";
        var commandName = "DeleteLifecyclePolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteLifecyclePolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteLifecyclePolicyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteLifecyclePolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DeleteLifecyclePolicyCommand(input, context);
    };
    DeleteLifecyclePolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DeleteLifecyclePolicyCommand(output, context);
    };
    return DeleteLifecyclePolicyCommand;
}($Command));
export { DeleteLifecyclePolicyCommand };
