// @aws-sdk/client-ecr/dist/es/commands/PutImageCommand.js
import { __extends } from "tslib";
import { PutImageRequest, PutImageResponse } from "../models/models_0";
import { deserializeAws_json1_1PutImageCommand, serializeAws_json1_1PutImageCommand } from "../protocols/Aws_json1_1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Creates or updates the image manifest and tags associated with an image.</p>
 *         <p>When an image is pushed and all new image layers have been uploaded, the PutImage API
 *             is called once to create or update the image manifest and the tags associated with the
 *             image.</p>
 *
 *         <note>
 *             <p>This operation is used by the Amazon ECR proxy and is not generally used by
 *         customers for pulling and pushing images. In most cases, you should use the <code>docker</code> CLI to pull, tag, and push images.</p>
 *          </note>
 */
var PutImageCommand = /** @class */ (function (_super) {
    __extends(PutImageCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function PutImageCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    PutImageCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ECRClient";
        var commandName = "PutImageCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PutImageRequest.filterSensitiveLog,
            outputFilterSensitiveLog: PutImageResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PutImageCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1PutImageCommand(input, context);
    };
    PutImageCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1PutImageCommand(output, context);
    };
    return PutImageCommand;
}($Command));
export { PutImageCommand };
