// @aws-sdk/client-ecr/dist/es/commands/PutImageScanningConfigurationCommand.js
import { __extends } from "tslib";
import { PutImageScanningConfigurationRequest, PutImageScanningConfigurationResponse } from "../models/models_0";
import { deserializeAws_json1_1PutImageScanningConfigurationCommand, serializeAws_json1_1PutImageScanningConfigurationCommand, } from "../protocols/Aws_json1_1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Updates the image scanning configuration for the specified repository.</p>
 */
var PutImageScanningConfigurationCommand = /** @class */ (function (_super) {
    __extends(PutImageScanningConfigurationCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function PutImageScanningConfigurationCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    PutImageScanningConfigurationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ECRClient";
        var commandName = "PutImageScanningConfigurationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PutImageScanningConfigurationRequest.filterSensitiveLog,
            outputFilterSensitiveLog: PutImageScanningConfigurationResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PutImageScanningConfigurationCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1PutImageScanningConfigurationCommand(input, context);
    };
    PutImageScanningConfigurationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1PutImageScanningConfigurationCommand(output, context);
    };
    return PutImageScanningConfigurationCommand;
}($Command));
export { PutImageScanningConfigurationCommand };
